gammapy/datasets/io.py

```diff
@@ -472,3 +477,167 @@ def read(self):
+        file_id : int
+            File index (last number of the fits file names). Default is 0.
+        edisp_bins : int
+            Number of margin bins to slice in energy. Default is 0.
+            If fermipy was configured with edisp_bins=0, it should be set to a value
+            edisp_bins>0 here in order to apply the energy dispersion correclty.
+        name : str, optional
+            Dataset name. The default is None, and the name is randomly generated.
+
+        Returns
+        -------
+        dataset : `~gammapy.datasets.MapDataset`
+            Map dataset.
+
+        """
+        from gammapy.datasets import MapDataset
+
+        path = Path(path)
+        counts = Map.read(path / f"ccube_0{str(file_id)}.fits")
+        exposure = Map.read(path / f"bexpmap_0{str(file_id)}.fits")
+        psf = PSFMap.read(path / f"psf_0{str(file_id)}.fits", format="gtpsf")
+        edisp = EDispKernelMap.read(path / f"drm_0{str(file_id)}.fits", format="gtdrm")
+
+        # check that fermipy edisp_bins are matching between edisp and exposure
+        edisp_axes = edisp.edisp_map.geom.axes
+        if (
+            len(edisp_axes["energy_true"].center)
+            != len(exposure.geom.axes["energy_true"].center) - 1
+        ):
+            raise ValueError(
+                "Energy true axes of exposure and DRM do not match. Check fermipy configuration."
+            )
+            edisp_axes = edisp.edisp_map.geom.axes
+
+        psf_r68s = psf.containment_radius(
+            0.68,
+            edisp_axes["energy_true"].center,
+            position=counts.geom.center_skydir,
+        )
+        # check that pdf is well defined (fails if edisp_bins>0 in fermipy)
+        if np.any(psf_r68s.value) == 0.0:
+            raise ValueError(
+                "PSF is not defined for all true energies. Check fermipy configuration."
+            )
+
+        # change counts energy axis unit keV->MeV
+        energy_axis = counts.geom.axes["energy"]._init_copy(
+            nodes=edisp_axes["energy"].edges
+        )
+        geom = counts.geom.to_image().to_cube([energy_axis])
+        counts = Map.from_geom(geom, data=counts.data)
+
+        # standardize dataset interpolating to same geom and axes
+        dataset = MapDataset(
+            counts=counts,
+            exposure=exposure,
+            psf=psf,
+            edisp=edisp,
+            name=name,
+        )
+        dataset = create_map_dataset_from_dl4(
+            dataset, geom=counts.geom, name=dataset.name
+        )
+
+        if edisp_bins > 0:  # slice edisp_bins
+            dataset = dataset.slice_by_idx(
+                dict(energy=slice(edisp_bins, -edisp_bins)), name=dataset.name
+            )
+
+        if isotropic_file:
+            model = create_fermi_isotropic_diffuse_model(
+                isotropic_file, datasets_names=[dataset.name]
+            )
+            dataset.models = Models([model])
+        return dataset
+
+    def read(self):
```

Comment:
Isnt it more reasonable to have the filename passed on read, rather than on init?

Reply:
I used the same logic than for the `OGIPDatasetReader`